src/bootstrap/check.rs

```diff
@@ -140,7 +140,10 @@ impl Step for Std {
             cargo_subcommand(builder.kind),
         );
 
-        cargo.arg("--all-targets");
+        if compiler.stage == 0 {
```

Comment:
Please add a comment here; adapting my explanation on the issue is fine, but it's very unclear right now why stage 0 is special cased.